pkg/objectcache/applicationprofilecache/applicationprofilecache.go

```diff
@@ -47,34 +48,113 @@ func newApplicationProfileState(ap *v1beta1.ApplicationProfile) applicationProfi
-	storageClient    versioned.SpdxV1beta1Interface
-	allProfiles      mapset.Set[string] // cache all the application profiles that are ready. this will enable removing from cache AP without pods that are running on the same node
-	nodeName         string
-	maxDelaySeconds  int // maximum delay in seconds before getting the full object from the storage
+	containerToSlug     maps.SafeMap[string, string]                      // cache the containerID to slug mapping, this will enable a quick lookup of the application profile
+	slugToAppProfile    maps.SafeMap[string, *v1beta1.ApplicationProfile] // cache the application profile
+	slugToContainers    maps.SafeMap[string, mapset.Set[string]]          // cache the containerIDs that belong to the application profile, this will enable removing from cache AP without pods
+	slugToState         maps.SafeMap[string, applicationProfileState]     // cache the containerID to slug mapping, this will enable a quick lookup of the application profile
+	storageClient       versioned.SpdxV1beta1Interface
+	allProfiles         mapset.Set[string] // cache all the application profiles that are ready. this will enable removing from cache AP without pods that are running on the same node
+	nodeName            string
+	maxDelaySeconds     int // maximum delay in seconds before getting the full object from the storage
+	userManagedProfiles maps.SafeMap[string, *v1beta1.ApplicationProfile]
 }
 
 func NewApplicationProfileCache(nodeName string, storageClient versioned.SpdxV1beta1Interface, maxDelaySeconds int) *ApplicationProfileCacheImpl {
-	return &ApplicationProfileCacheImpl{
-		nodeName:         nodeName,
-		maxDelaySeconds:  maxDelaySeconds,
-		storageClient:    storageClient,
-		containerToSlug:  maps.SafeMap[string, string]{},
-		slugToContainers: maps.SafeMap[string, mapset.Set[string]]{},
-		allProfiles:      mapset.NewSet[string](),
+	cache := &ApplicationProfileCacheImpl{
+		nodeName:            nodeName,
+		maxDelaySeconds:     maxDelaySeconds,
+		storageClient:       storageClient,
+		containerToSlug:     maps.SafeMap[string, string]{},
+		slugToAppProfile:    maps.SafeMap[string, *v1beta1.ApplicationProfile]{},
+		slugToContainers:    maps.SafeMap[string, mapset.Set[string]]{},
+		slugToState:         maps.SafeMap[string, applicationProfileState]{},
+		allProfiles:         mapset.NewSet[string](),
+		userManagedProfiles: maps.SafeMap[string, *v1beta1.ApplicationProfile]{},
 	}
 
+	return cache
 }
 
 // ------------------ objectcache.ApplicationProfileCache methods -----------------------
 
+func (ap *ApplicationProfileCacheImpl) handleUserManagedProfile(appProfile *v1beta1.ApplicationProfile) {
+	baseProfileName := strings.TrimPrefix(appProfile.GetName(), "ug-")
+	baseProfileUniqueName := objectcache.UniqueName(appProfile.GetNamespace(), baseProfileName)
+
+	// Get the full user managed profile from the storage
+	fullAP, err := ap.getApplicationProfile(appProfile.GetNamespace(), appProfile.GetName())
```

Comment:
maybe rename to `userManagedProfile` like elsewere

Reply:
done